plugins/inputs/cisco_telemetry_mdt_netconf/cisco_telemetry_mdt_netconf.go

```diff
@@ -0,0 +1,941 @@
+	Receive(response interface{}) error
+	Close() error
+}
+
+// service represents an interface for NETCONF client services
+type service interface {
+	createService(context.Context, *CiscoTelemetryNETCONF)
+	createRequests(context.Context, *CiscoTelemetryNETCONF) error
+}
+
+// dialinSubscriptionRequestsService represents the service for
+// dial-in subscriptions established by the NETCONF client
+type dialinSubscriptionRequestsService struct {
+	Subscriptions []dialinSubscriptionRequest       `toml:"subscription"`
+	Notifications []notificationSubscriptionRequest `toml:"notification"`
+	*setting
+}
+
+// getRequestsService represents the service for get requests
+// made by the NETCONF client
+type getRequestsService struct {
+	Gets []getRequest `toml:"get"`
+	*setting
+}
+
+// dialinSubscriptionRequest given as input to the plugin
+type dialinSubscriptionRequest struct {
+	XPathFilter   string `toml:"xpath_filter"`
+	UpdateTrigger string `toml:"update_trigger"`
+	Period        config.Duration
+	Keys          []string
+}
+
+// notificationSubscriptionRequest given as input to the plugin
+type notificationSubscriptionRequest struct {
+	Stream string `toml:"stream"`
+	Keys   []string
+}
+
+// getRequest given as input to the plugin
+type getRequest struct {
+	SelectFilter string `toml:"xpath_filter"`
+	Period       config.Duration
+	Keys         []string
+}
+
+// netconfSubscriptionRequest sent through NETCONF, as per specification
+type netconfSubscriptionRequest struct {
+	XMLName         xml.Name `xml:"urn:ietf:params:xml:ns:yang:ietf-event-notifications establish-subscription"`
+	YangPush        string   `xml:"xmlns:yp,attr"`
+	Stream          string   `xml:"stream"`
+	XPathFilter     string   `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push xpath-filter"`
+	Period          uint64   `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push period,omitempty"`           // hundreds of a second 100=1s
+	DampeningPeriod *uint64  `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push dampening-period,omitempty"` // on-change
+}
+
+// netconfYangPush notification received through NETCONF, as per specification
+type netconfYangPush struct {
+	netconf.Notification
+	PushUpdate struct {
+		Content struct {
+			InnerXML []byte `xml:",innerxml"`
+		} `xml:"datastore-contents-xml"`
+	} `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push push-update"`
+}
+
+// netconfYangEvent notification received through NETCONF, as per specification
+type netconfYangEvent struct {
+	netconf.Notification
+	InnerXML []byte `xml:",innerxml"`
+}
+
+// CiscoTelemetryNETCONF plugin
+type CiscoTelemetryNETCONF struct {
+	ServerAddress string `toml:"server_address"`
+
+	Dsrs *dialinSubscriptionRequestsService `toml:"subscription_service"`
+	Grs  *getRequestsService                `toml:"get_service"`
+
+	// SSH connection - credentials
+	Username string
+	Password string
+
+	// SSH connection - security
+	IgnoreServerAuthenticity bool
+	ServerPublicKey          string
```

Comment:
Missing TOML tags

Reply:
Thank you. Addressed in commit https://github.com/cisco-ie/telegraf/commit/c9d4737a76e143d1634ed83341831e7db98225d8.